src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/MintermClassifier.cs

```diff
@@ -20,81 +20,124 @@ namespace System.Text.RegularExpressions.Symbolic
     /// </remarks>
     internal sealed class MintermClassifier
     {
-        /// <summary>An array used when there's a single minterm, in order to map every ASCII character to it trivially.</summary>
-        private static readonly int[] AllAsciiIsZeroMintermArray = new int[128];
+        /// <summary>An array used to map characters to minterms</summary>
+        private readonly byte[]? _lookup;
 
-        /// <summary>Array providing fast mapping from an ASCII character (the array index) to its corresponding minterm ID.</summary>
-        private readonly int[] _ascii;
-        /// <summary>A multi-terminal BDD for mapping any non-ASCII character to its associated minterm ID.</summary>
-        /// <remarks>
-        /// The use of a multi-terminal BDD here is an implementation detail.  Should we decide its important to optimize non-ASCII inputs further,
-        /// or to consolidate the mechanism with the other engines, an alternatie lookup algorithm / data structure could be employed.
-        /// </remarks>
-        private readonly BDD _nonAscii;
+        /// <summary>
+        /// Fallback lookup if over 255 minterms. This is rarely used.
+        /// </summary>
+        private readonly int[]? _intLookup;
+
```

Comment:
```suggestion
```

Reply:
also is this rarely used part also covered by the extra test you added?

Reply:
Only the pattern at https://github.com/dotnet/runtime/blob/4f96b8f541a62245eb5ca4b643363785cc0d58bf/src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexPcreTests.cs#L391 reaches this case.

i could add an extra test case just to be sure

Reply:
Or maybe it was this one instead:
https://github.com/dotnet/runtime/blob/4f96b8f541a62245eb5ca4b643363785cc0d58bf/src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexPcreTests.cs#L2258

There was definitely a case in RegexPcreTests which reached this path

Reply:
(none of those note they hit this >255 path)

incidentally this may be interesting if you wish to try it.  at least for locally checking block coverage.
https://github.com/dotnet/runtime/blob/a42b063686e9e4c263e6a773b5836a594a62343b/docs/workflow/building/libraries/code-coverage.md#L1

Reply:
I will add a quick unit test for it as well just to be sure, then it's more certain